graphdatascience/procedure_surface/api/wcc_endpoints.py

```diff
@@ -0,0 +1,276 @@
+        consecutive_ids: Optional[bool] = None,
+        relationship_weight_property: Optional[str] = None,
+        write_concurrency: Optional[Any] = None,
+    ) -> WccWriteResult:
+        """
+        Executes the WCC algorithm and writes the results to the Neo4j database.
+
+        Parameters
+        ----------
+        G : Graph
+            The graph to run the algorithm on
+        write_property : str
+            The property name to write component IDs to
+        min_component_size : Optional[int], default=None
+            Don't write components with fewer nodes than this
+        threshold : Optional[float], default=None
+            The minimum required weight to consider a relationship during traversal
+        relationship_types : Optional[List[str]], default=None
+            The relationship types to project
+        node_labels : Optional[List[str]], default=None
+            The node labels to project
+        sudo : Optional[bool], default=None
+            Run analysis with admin permission
+        log_progress : Optional[bool], default=None
+            Whether to log progress
+        username : Optional[str], default=None
+            The username to attribute the procedure run to
+        concurrency : Optional[Any], default=None
+            The number of concurrent threads
+        job_id : Optional[Any], default=None
+            An identifier for the job
+        seed_property : Optional[str], default=None
+            Defines node properties that are used as initial component identifiers
+        consecutive_ids : Optional[bool], default=None
+            Flag to decide whether component identifiers are mapped into a consecutive id space
+        relationship_weight_property : Optional[str], default=None
+            The property name that contains weight
+        write_concurrency : Optional[Any], default=None
+            The number of concurrent threads during the write phase
+
+        Returns
+        -------
+        WccWriteResult
+            Algorithm metrics and statistics
+        """
+        pass
+
+
+@dataclass(frozen=True, repr=True)
+class WccMutateResult:
+    component_count: int
+    component_distribution: dict[str, Any]
+    pre_processing_millis: int
+    compute_millis: int
+    post_processing_millis: int
+    mutate_millis: int
+    node_properties_written: int
+    configuration: dict[str, Any]
+
+
+@dataclass(frozen=True, repr=True)
+class WccStatsResult:
+    component_count: int
+    component_distribution: dict[str, Any]
+    pre_processing_millis: int
+    compute_millis: int
+    post_processing_millis: int
+    configuration: dict[str, Any]
+
+
+@dataclass(frozen=True, repr=True)
+class WccWriteResult:
+    component_count: int
+    component_distribution: dict[str, Any]
+    pre_processing_millis: int
+    compute_millis: int
+    write_millis: int
+    post_processing_millis: int
+    node_properties_written: int
+    configuration: dict[str, Any]
```

Comment:
We could consider some static code reuse scheme here, like inheritance, for shared fields.
Or we just have good tests to make sure we don't misspell or forget any field.
We used class inheritance for these classes in Java, but while it gave us type safety it does convolute the code when each level just adds one or two fields and one has to navigate to super classes.
The latest approach of using Java `record`s that implement Java `interface`s to make sure that shared field names are the same and are present is one I like quite a lot. I wonder if we can do a similar thing in Python.